### PR TITLE
Sorting2

### PR DESCRIPTION
**Pull request by datawhores**
Changes
- added the ability to make sort directory on the fly, since were only moving files into it
- add the ability to set a separate scan directory, otherwise  just use the download_dir
- add progress bar for sorting folders
- disable database scan if sorting all files, and not just cdl

### CODE REVIEW

**Comment by datawhores** [comment]
It would have been okay to increment to 5.5 in when sorted was added, because
it more then just a bug fix or adding a new  website

**Comment by jbsparrow** [comment]
I'll keep that in mind for future updates, never really worked with building a package before this so I'm not the most well-versed on how versioning works. Thank you.

**Review by jbsparrow — CHANGES_REQUESTED** [review]
Please add update info to the changelog 🙏

File: cyberdrop_dl/utils/sorting.py [hunk 1]
```diff
@@ -88,55 +91,63 @@ async def sort(self) -> None:
...
         unique_download_paths = await self.db_manager.history_table.get_unique_download_paths()
-        download_folders = [Path(download_path[0]) for download_path in unique_download_paths if Path(download_path[0]).is_dir() and Path(download_path[0]) != self.download_dir]
+        download_folders = download_folders = [Path(download_path[0]) for download_path in unique_download_paths if Path(download_path[0]).is_dir() and Path(download_path[0]) != self.download_dir]
         existing_folders = []
-        
         for folder in download_folders:
             try:
                 relative_folder = folder.relative_to(self.download_dir)
                 base_folder = self.download_dir / relative_folder.parts[0]
+                print(relative_folder)
```
Comment (jbsparrow) [inline cyberdrop_dl/utils/sorting.py:140]: Is this intended to be included in the PR?

File: cyberdrop_dl/utils/sorting.py [hunk 2]
```diff
@@ -88,55 +91,63 @@ async def sort(self) -> None:
...
+        await log_with_color(f"Organized: {self.video_count} Video Files", "green", 20)
+        await log_with_color(f"Organized: {self.other_count} Other Files", "green", 20)
+
+    async def get_download_folder(self):
+        """Gets the download folder"""
+        if not self.sort_cdl_only:
+            return []
         unique_download_paths = await self.db_manager.history_table.get_unique_download_paths()
-        download_folders = [Path(download_path[0]) for download_path in unique_download_paths if Path(download_path[0]).is_dir() and Path(download_path[0]) != self.download_dir]
+        download_folders = download_folders = [Path(download_path[0]) for download_path in unique_download_paths if Path(download_path[0]).is_dir() and Path(download_path[0]) != self.download_dir]
```
Comment (jbsparrow) [inline cyberdrop_dl/utils/sorting.py:134]: Defining download_folders twice

**Review by jbsparrow — CHANGES_REQUESTED** [review]

File: cyberdrop_dl/utils/sorting.py (hunk 1 above)
Comment (datawhores) [reply cyberdrop_dl/utils/sorting.py:140]: no that must have been from testing

**Review by jbsparrow — APPROVED** [review]

**Review by jbsparrow — APPROVED** [review]

**Comment by jbsparrow** [comment]
> > I'll keep that in mind for future updates, never really worked with building a package before this so I'm not the most well-versed on how versioning works. Thank you.
> 
> 
> 
> It is usually 
> 
> - bug fixes are 1.0.0  to 1.0.1 type increments or patch releases
> 
> - new features are 1.0.0  to 1.1.0 type increments or minor releases
> 
> - the last one I find harder to define but it is  1.0 to 2.0 type increments or major release
> 
> it usually meant for larger changes or ones that are not backwards compatible
> 
> last time  that happened Jules made 100s of commits, and did a full-rewrite
> 
> https://github.com/Jules-WinnfieldX/CyberDropDownloader/compare/4.2.231...5.0.1

Thank you!! :))

**Pull request merged**